zenoh/src/net/routing/hat/router/token.rs

```diff
@@ -406,9 +412,26 @@ fn propagate_forget_simple_token(
                     res.expr(),
                 ),
             );
-        } else if face_hat!(face).remote_interests.values().any(|(r, o)| {
-            o.tokens() && r.as_ref().map(|r| r.matches(res)).unwrap_or(true) && !o.aggregate()
-        }) {
+        // NOTE(fuzzypixelz): We need to check that `face` is not the source Face of the token
+        // undeclaration, otherwise the undeclaration would be duplicated at the source Face.
+        //
+        // The code path that leads to `src_face` being `None` originate in:
+        //   - `HatBaseTrait::closing`
+        //   - `HatBaseTrait::handle_oam`
+        //   - `undeclare_client_token`: when a peer/client undeclares a token and no other
+        //     peer/client has declared it (or has undeclared it) then **this** router's propagation
+        //     of the token is undeclared. In this case we don't have a source Face (there is no
+        //     "self" face).
+        //
+        // In these cases we don't have access to a Face as we don't received an undeclaration and
+        // we default to true.
+        //
+        // TODO: Could we use ZIDs instead of Faces?
```

Comment:
Why would ZIDs be better ? It seems to me that comparing faces ids is faster.

Reply:
My reasoning was that we always have ZIDs, even when there is no source Face.